### ISSUE TITLE
[Logstash]: Conflicting output configuration information for `elastic-agent-pipeline.conf` under docs.

### ISSUE THREAD

**Issue by amolnater-qasource** [elastic/observability-docs#1850]
**Doc Link:**
https://www.elastic.co/guide/en/fleet/8.2/secure-logstash-connections.html#configure-ls-ssl

**Description:**
Conflicting information for configuration of output under `elastic-agent-pipeline.conf` for Elasticsearch service example.
- It shows that elasticsearch- cloud id is required for the cloud environments.
- However as per Settings>Output>Type: Logstash>View Steps- it shows hosts

**Screenshots:**
![3](https://user-images.githubusercontent.com/77374876/166659584-bf168ab2-cbe4-494a-8be7-ff19c1b6aca2.JPG)
![4](https://user-images.githubusercontent.com/77374876/166659594-45fba648-f261-45da-b8f3-9969ee705018.JPG)

**Related ticket:**
https://github.com/elastic/observability-docs/issues/1691

**Comment by dikshachauhan-qasource** [comment]
Secondary review is done

**Comment by amolnater-qasource** [comment]
FYI @nchaulet

**Comment by dedemorton** [comment]
I show cloud_id in the docs example because I thought that was the preferred way to connect to ESS. If that's correct, then I'd argue that this is a product bug rather than a doc bug. (The inline instructions should show cloud_id when Kibana is running on cloud.)

@nchaulet WDYT? Maybe someone on the cloud or logstash dev team should weigh in here. Who should we ask?

**Comment by jsvd** [comment]
> It shows that elasticsearch- cloud id is required for the cloud environments.

`cloud_id` is indeed recommended for cloud environments (not required), and `hosts` can be used as well. I believe this is just a current limitation where the guide doesn't detect that it's running on an Elastic cloud deployment.

We can't default to cloud_id as the cluster may be on prem, but I'm +1 on having this distinction: show `hosts` on prem and `cloud_id` on SaaS (stretch goal: populate the field with actual cloud id).

**Comment by dedemorton** [comment]
Should I transfer this issue to Kibana, or does someone want to create a new issue to track the change that's required here?

**Comment by nchaulet** [comment]
@dedemorton Yes please transfer it to Kibana, so we will have all the context
